x-pack/filebeat/module/zeek/notice/config/notice.yml

```diff
@@ -0,0 +1,113 @@
+ - {{$path}}
+{{ end }}
+exclude_files: [".gz$"]
+tags: {{.tags}}
+
+json.keys_under_root: false
+
+processors:
+  - drop_fields:
+      fields: ["json.actions"]
+  - rename:
+      fields:
+        - from: "json"
+          to: "zeek.notice"
+
+        - from: "zeek.notice.src"
+          to: "source.address"
+          when:
+            has_fields: ["zeek.notice.src"]
+
+        - from: "zeek.notice.dest"
+          to: "destination.address"
+          when:
+            has_fields: ["zeek.notice.dest"]
+
+        - from: "zeek.notice.uid"
+          to: "zeek.session_id"
+          when:
+            has_fields: ["zeek.notice.uid"]
+
+        - from: "zeek.notice.p"
+          to: "destination.port"
+          when:
+            has_fields: ["zeek.notice.p"]
+
+        - from: "zeek.notice.conn"
+          to: "zeek.notice.connnection_id"
+          when:
+            has_fields: ["zeek.notice.conn"]
+
+        - from: "zeek.notice.iconn"
+          to: "zeek.notice.icmp_id"
+          when:
+            has_fields: ["zeek.notice.iconn"]
+
+        - from: "zeek.notice.id.orig_h"
+          to: "source.address"
+          when:
+            has_fields: ["zeek.notice.id.orig_h"]
+
+        - from: "zeek.notice.id.orig_p"
+          to: "source.port"
+          when:
+            has_fields: ["zeek.notice.id.orig_p"]
+
+        - from: "zeek.notice.id.resp_h"
+          to: "destination.address"
+          when:
+            has_fields: ["zeek.notice.id.resp_h"]
+
+        - from: "zeek.notice.id.resp_p"
+          to: "destination.port"
+          when:
+            has_fields: ["zeek.notice.id.resp_p"]
+
+        - from: "zeek.notice.proto"
+          to: "network.transport"
+          when:
+            has_fields: ["zeek.notice.proto"]
+
+        - from: "zeek.notice.id.orig_p"
+          to: "source.port"
+          when:
+            has_fields: ["zeek.notice.id.orig_p"]
+
+        - from: "zeek.notice.f.id"
+          to: "zeek.notice.file.id"
+          when:
+            has_fields: ["zeek.notice.f.id"]
+
+        - from: "zeek.notice.f.parent_id"
+          to: "dzeek.notice.file.parent_id"
+          when:
+            has_fields: ["zeek.notice.f.parent_id"]
+
+        - from: "zeek.notice.f.source"
+          to: "zeek.notice.file.source"
+          when:
+            has_fields: ["zeek.notice.f.source"]
+
+        - from: "zeek.notice.f.is_orig"
+          to: "zeek.notice.file.is_orig"
+          when:
+            has_fields: ["zeek.notice.f.is_orig"]
+
+        - from: "zeek.notice.f.seen_bytes"
+          to: "zeek.notice.file.seen_bytes"
+          when:
+            has_fields: ["zeek.notice.f.seen_bytes"]
+
+        - from: "zeek.notice.f.total_bytes"
+          to: "zeek.notice.file.total_bytes"
+          when:
+            has_fields: ["zeek.notice.f.total_bytes"]
+
+      ignore_missing: true
+      fail_on_error: false
+
+  - drop_fields:
+      fields: ["zeek.notice.remote_location", "zeek.notice.f"]
```

Comment:
Should these processors go to the pipeline? It is a bit confusing to have some processing in filebeat and some in the ingest pipeline.

Reply:
The original intend was to reduce workload in the ES ingest nodes by distributing it to beats side. But it seems to cause more confusion. I will move the processing to ingest node now.

Reply:
Also, is "zeek.notice.f" a typo, or is `f` an actual field name?

Reply:
f is an actually field in zeek notice. Some of its sub-fields are normalized to a better field name and the f field itself is dropped.

Reply:
@webmat yeah, I agree that it'd be nice to do more things in filebeat and less in the ingest pipelines.

But having some processing in filebeat and some in the pipelines is a bit confusing, as we are splitting some "logic" between two separated things. Also when investigating issues users (or us) have to remember that some things are being done by processors and some in the pipelines (if installed).

At the moment there are things that are only going to work in ingest pipelines, and I don't think we want to have yet another implementation for them in filebeat.

Maybe we should define some clear guidelines to decide what to place in processors and what in pipelines. For example renaming fields obtained from json logs like here can make sense in processors, but I'd leave parsing for example for ingest nodes.